sympy/solvers/solveset.py

```diff
@@ -304,6 +305,14 @@ def inv(trig):
                 invs += Union(*[imageset(Lambda(n, L(g)), S.Integers) for g in g_ys])
             return _invert_real(f.args[0], invs, symbol)
 
+    if isinstance(f, LambertW):
+        return _invert_real(f.args[0], imageset(Lambda(n, n*exp(n)), g_ys), symbol)
+
+    if _is_lambert(f - g_ys.args[0], symbol):
+        s = _solve_as_lambert(f, g_ys.args[0], symbol, domain=S.Reals)
+        if s.has(LambertW):
+            return symbol, s
```

Comment:
Would `_solve_as_lambert` ever return something that does not have a `LambertW` in it? If this is really checking for `ConditionSet` then it would be best to check for `ConditionSet`.

Why does `_solve_as_lambert` bother creating a `ConditionSet` if it is never used anywhere?

Reply:
What would this do if the `ConditionSet` had a `LambertW` in it?

Reply:
If this is really checking for ConditionSet then it would be best to check for ConditionSet.

Yes, it returns `ConditionSet` in some of the cases.

Reply:
> What would this do if the ConditionSet had a LambertW in it?

Oh, I'll make a case for `ConditionSet` instead!
`if not isinstance(s,ConditionSet):`
Will this be a good one?

Reply:
Yes, but actually there's no point in creating the conditionset in the first place. The function could just return None instead.

Reply:
```
if _is_lambert(f - g_ys.args[0], symbol):
        soln = _solve_as_lambert(f, g_ys.args[0], symbol, domain=S.Reals)
        if soln:
            return symbol, soln
```